cmd/thermal-recorder/cptv_test.go

```diff
@@ -0,0 +1,150 @@
+	}
+
+	assert.Equal(t, expectedResults, results)
+}
+
+func TestCptvNoiseRecordings(t *testing.T) {
+	config := CurrentConfig()
+
+	results := NewCPTVPlaybackTester(config).TestAllCPTVFiles(GetBaseDir() + "/motiontest/noise")
+	expectedResults := []string{
+		"noise_01.cptv        Detected: None             Recorded: None             Motion frames: 1/177",
+		"noise_02.cptv        Detected: None             Recorded: None             Motion frames: 1/99",
+		"noise_03.cptv        Detected: (75:79)          Recorded: (76:88)          Motion frames: 9/117",
+		"noise_05.cptv        Detected: (19:76)(90:94)   Recorded: (20:85)(91:103)  Motion frames: 43/119",
+		"skyline.cptv         Detected: None             Recorded: None             Motion frames: 1/91",
+	}
+
+	assert.Equal(t, expectedResults, results)
+}
+
+// DoTestResearchAnimalRecordings - change this to test to run though different scenarios of test
+// calculations.   It will output the results to /motiontest/results
+func DoTestResearchAnimalRecordings(t *testing.T) {
+	testname := "cut off - noise"
+	searchDir := GetBaseDir() + "/motiontest/noise"
+
+	f, err := os.Create(GetBaseDir() + "/motiontest/results/" + testname)
+	if err != nil {
+		return
+	}
+	defer f.Close()
+	writer := bufio.NewWriter(f)
+
+	config := CurrentConfig()
+	config.Motion.TempThresh = 2900
+	ExperiementAndWriteResultsToFile(testname+"2900", config, searchDir, writer)
+
+	config.Motion.TempThresh = 2800
+	ExperiementAndWriteResultsToFile(testname+"2800", config, searchDir, writer)
+
+	config.Motion.TempThresh = 2700
+	ExperiementAndWriteResultsToFile(testname+"2700", config, searchDir, writer)
+
+	config.Motion.TempThresh = 2500
+	ExperiementAndWriteResultsToFile(testname+"2500", config, searchDir, writer)
+
+	ExperiementAndWriteResultsToFile("Current config", CurrentConfig(), searchDir, writer)
+
+	ExperiementAndWriteResultsToFile("Old default", OldDefaultConfig(), searchDir, writer)
+}
+
+func ExperiementAndWriteResultsToFile(name string, config *Config, dir string, writer *bufio.Writer) {
+	fmt.Fprintf(writer, "Results for %s", name)
+	fmt.Fprintln(writer)
+
+	results := NewCPTVPlaybackTester(config).TestAllCPTVFiles(dir)
+
+	fmt.Fprintf(writer, "%-10s:  %ds - %ds", "Recording limits", config.MinSecs, config.MaxSecs)
+	fmt.Fprintln(writer, "")
+	fmt.Fprintln(writer, config.Motion)
+
+	for ii := range results {
+		fmt.Fprintln(writer, results[ii])
+	}
+	fmt.Fprintln(writer)
+	fmt.Fprintln(writer)
+
+	writer.Flush()
+}
+
+func GetBaseDir() string {
+	_, file, _, _ := runtime.Caller(0)
+
+	dir, _ := filepath.Abs(filepath.Dir(file))
+
+	return dir
+}
+
+func BenchmarkMotionDetection(b *testing.B) {
+	config := CurrentConfig()
+
+	tester := NewCPTVPlaybackTester(config)
+	frames := tester.LoadAllCptvFrames(GetBaseDir() + "/motiontest/animals/recalc.cptv")
+
+	listener := new(HardwareListener) // currently doesn't do anything
```

Comment:
Although the HardwareListener doesn't do anything, there's no need for the comment. Also, the comment becomes incorrect as soon as the HardwareListener changes.